base_layer/core/src/transaction.rs

```diff
@@ -503,9 +503,15 @@ impl Transaction {
         &self.body
     }
 
+    /// Returns the byte size or weight of a transaction
     pub fn get_weight(&self) -> Result<usize, MessageFormatError> {
```

Comment:
This isn't quite right.

The weight calculation is actually in [fee.rs#L35], though it's still bundles in with the fee calculation.

We should just extract that out; have
```
pub fn calculate_weight(num_inputs: usize, num_outputs: usize) -> u64 {
          COST_PER_INPUT * num_inputs as u64 + COST_PER_OUTPUT * num_outputs as u64
    }
```

Reply:
I wondered if there wasn't a formula. Change applied, but I hope I interpreted your suggestion correctly.